src/main/java/com/timgroup/statsd/CgroupReader.java

```diff
@@ -70,4 +116,84 @@ public static String parse(final String cgroupsContent) {
 
         return null;
     }
+
+    /**
+     * Returns true if the host cgroup namespace is used.
+     * It looks at the inode of `/proc/self/ns/cgroup` and compares it to
+     * HOST_CGROUP_NAMESPACE_INODE.
+     * 
+     * @param path Path to the cgroup namespace file.
+     */
+    private static boolean isHostCgroupNamespace(final Path path) {
+        long hostCgroupInode = inodeForPath(path);
+        return hostCgroupInode == HOST_CGROUP_NAMESPACE_INODE;
+    }
+
+    /**
+     * Returns the inode for the given path.
+     * 
+     * @param path Path to the cgroup namespace file.
+     */
+    private static long inodeForPath(final Path path) {
+        try {
+            long inode = (long) Files.getAttribute(path, "unix:ino");
+            return inode;
+        } catch (Exception e) {
+            return 0;
+        }
+    }
+
+    /**
+     * Returns the cgroup controller inode for the given cgroup mount path and
+     * procSelfCgroupPath.
+     * 
+     * @param cgroupMountPath Path to the cgroup mount point.
+     * @param cgroupContent   String content of the cgroup file.
+     */
+    public static String getCgroupInode(final Path cgroupMountPath, final String cgroupContent) throws IOException {
+        Map<String, String> cgroupControllersPaths = parseCgroupNodePath(cgroupContent);
+        if (cgroupControllersPaths == null) {
+            return null;
+        }
+
+        // Retrieve the cgroup inode from /sys/fs/cgroup+controller+cgroupNodePath
+        List<String> controllers = Arrays.asList(CGROUPV1_BASE_CONTROLLER, CGROUPV2_BASE_CONTROLLER);
+        for (String controller : controllers) {
+            String cgroupNodePath = cgroupControllersPaths.get(controller);
+            if (cgroupNodePath == null) {
+                continue;
+            }
+            Path path = Paths.get(cgroupMountPath.toString(), controller, cgroupNodePath);
+            long inode = inodeForPath(path);
+            if (inode > 2) {
+                return "in-" + inode;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns a map of cgroup controllers and their corresponding cgroup path.
+     * 
+     * @param cgroupContent Cgroup file content.
+     */
+    public static Map<String, String> parseCgroupNodePath(final String cgroupContent) throws IOException {
+        Map<String, String> res = new HashMap<>();
+        BufferedReader br = new BufferedReader(new StringReader(cgroupContent));
+
+        String line;
+        while ((line = br.readLine()) != null) {
+            String[] tokens = line.split(":");
+            if (tokens.length != 3) {
+                continue;
+            }
+            if (CGROUPV1_BASE_CONTROLLER.equals(tokens[1]) || tokens[1].isEmpty()) {
+                res.put(tokens[1], tokens[2]);
```

Comment:
Would it make sense to combine this method with `getCgroupInode`? cgroup versions are mutually exclusive so the order in which we check for them does not matter, and the map of paths is not used anywhere else.

```suggestion
                Path path = Paths.get(cgroupMountPath.toString(), controller, cgroupNodePath);
                long inode = inodeForPath(path);
                if (inode > 2) {
                    return "in-" + inode;
                }
```

Reply:
Actually there is a `unified` cgroup version that can use both. I think it would make sense to keep this implementation that first tries the Cgroupv1 controller then tries with the Cgroupv2 one.